pkg/controller/wip/controller.go

```diff
@@ -17,54 +17,61 @@ limitations under the License.
 package wip
 
 import (
+	"fmt"
+
 	"github.com/golang/glog"
 
 	"k8s.io/client-go/1.5/kubernetes"
+	"k8s.io/client-go/1.5/pkg/api"
 	"k8s.io/kubernetes/pkg/client/cache"
 	"k8s.io/kubernetes/pkg/util/runtime"
 
 	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
-	servicecatalogclientset "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	servicecatalogclientset "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset/typed/servicecatalog/v1alpha1"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers/servicecatalog/v1alpha1"
+	listers "github.com/kubernetes-incubator/service-catalog/pkg/client/listers/servicecatalog/v1alpha1"
 )
 
 // NewController returns a new Open Service Broker catalog
 // controller.
 func NewController(
 	kubeClient kubernetes.Interface,
-	serviceCatalogClient servicecatalogclientset.Interface,
-	brokerInformer cache.SharedInformer,
-	serviceClassInformer cache.SharedInformer,
-	instanceInformer cache.SharedInformer,
-	bindingInformer cache.SharedInformer,
+	serviceCatalogClient servicecatalogclientset.ServicecatalogV1alpha1Interface,
+	brokerInformer informers.BrokerInformer,
+	serviceClassInformer informers.ServiceClassInformer,
+	instanceInformer informers.InstanceInformer,
+	bindingInformer informers.BindingInformer,
+	brokerClientCreateFunc brokerapi.CreateFunc,
 ) (Controller, error) {
+	serviceClassLister := serviceClassInformer.Lister()
+
 	controller := &controller{
-		kubeClient:           kubeClient,
-		serviceCatalogClient: serviceCatalogClient,
-		brokerInformer:       brokerInformer,
-		serviceClassInformer: serviceClassInformer,
-		instanceInformer:     instanceInformer,
-		bindingInformer:      bindingInformer,
+		kubeClient:             kubeClient,
+		serviceClassLister:     serviceClassLister,
+		serviceCatalogClient:   serviceCatalogClient,
+		brokerClientCreateFunc: brokerClientCreateFunc,
 	}
 
-	brokerInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+	brokerInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
The naming of the types themselves is a little awkward.  The types are named 'Informer', but have the following interface methods:

- `Lister` - which gives you a lister
- `Informer` - which gives you a `SharedInformer` that you can add event handlers to

So, what you're seeing is idiomatic and correct, but awkward.